dags/civicrm_email_report_import_pipeline.py

```diff
@@ -0,0 +1,126 @@
+# Note: DagBag.process_file skips files without "airflow" or "DAG" in them
+
+import os
+import logging
+from datetime import timedelta
+
+from data_pipeline.utils.dags.data_pipeline_dag_utils import create_dag, create_python_task
+from data_pipeline.utils.pipeline_file_io import get_yaml_file_as_dict
+from data_pipeline.civicrm_email_report.civicrm_email_report import iter_email_report
+from data_pipeline.utils.data_store.bq_data_service import (
+    get_distinct_values_from_bq,
+    load_given_json_list_data_from_tempdir_to_bq
+)
+from data_pipeline.civicrm_email_report.get_civicrm_email_report_data_config import (
+    CiviCrmEmailReportDataConfig
+)
+from data_pipeline.utils.pipeline_file_io import read_file_content
+
+LOGGER = logging.getLogger(__name__)
+
+CIVICRM_KEY_SECRET_FILE_PATH_ENV = "CIVICRM_KEY_SECRET_FILE_PATH"
+
+CIVICRM_API_KEY_FILE_PATH_ENV = "CIVICRM_API_KEY_FILE_PATH"
+CIVICRM_SITE_KEY_FILE_PATH_ENV = "CIVICRM_SITE_KEY_FILE_PATH"
+
+CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME = "CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH"
+
+DEPLOYMENT_ENV_ENV_NAME = "DEPLOYMENT_ENV"
+DEFAULT_DEPLOYMENT_ENV = "ci"
+
+DAG_ID = "Civicrm_Email_Summary_Report_Pipeline"
+CIVICRM_EMAIL_REPORT_DATA_PIPELINE_SCHEDULE_INTERVAL_ENV_NAME = (
+    "CIVICRM_EMAIL_REPORT_DATA_PIPELINE_SCHEDULE_INTERVAL"
+)
+
+
+def get_env_var_or_use_default(env_var_name, default_value=None):
+    return os.getenv(env_var_name, default_value)
+
+
+def get_data_config(**kwargs):
+    config_file_path = get_env_var_or_use_default(
+        CIVICRM_EMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME, ""
+    )
+    data_config_dictionary = get_yaml_file_as_dict(config_file_path)
+    kwargs["ti"].xcom_push(
+        key="data_config_dict",
+        value=data_config_dictionary
+    )
+
+
+def data_config_from_xcom(context):
+    dag_con = context["ti"]
+    data_config_dict = dag_con.xcom_pull(
+        key="data_config_dict",
+        task_ids="get_data_config"
+    )
+    deployment_env = get_env_var_or_use_default(
+        DEPLOYMENT_ENV_ENV_NAME,
+        DEFAULT_DEPLOYMENT_ENV
+    )
+    data_configuration = CiviCrmEmailReportDataConfig(
+        data_config_dict, deployment_env)
+    LOGGER.info('data_config: %r', data_configuration)
+    return data_configuration
+
+
+def get_civicrm_credential(key_name: str):
```

Comment:
minor: Maybe rather than passing in the key name, we could just pass in the env variable name?
Otherwise at the moment it isn't clear what the accepted key names are.

Reply:
Or alternatively these could just be two separate functions. They don't really share any code.

Reply:
used env var name